rewrite-gradle/src/main/java/org/openrewrite/gradle/RemoveGradleDependency.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2021 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.gradle;
+
+import lombok.AllArgsConstructor;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Option;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.gradle.IsBuildGradle;
+import org.openrewrite.gradle.util.Dependency;
+import org.openrewrite.gradle.util.DependencyStringNotationConverter;
+import org.openrewrite.groovy.GroovyVisitor;
+import org.openrewrite.groovy.tree.G;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.semver.DependencyMatcher;
+
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+@AllArgsConstructor
+class RemoveGradleDependency extends Recipe {
+
+    @Option(displayName = "The dependency configuration", description = "The dependency configuration to remove from.", example = "api", required = false)
+    @Nullable
+    String configuration;
+
+    @Option(displayName = "Group",
+            description = "The first part of a dependency coordinate `com.google.guava:guava:VERSION`. This can be a glob expression.",
+            example = "com.fasterxml.jackson*")
+    String groupId;
+
+    @Option(displayName = "Artifact",
+            description = "The second part of a dependency coordinate `com.google.guava:guava:VERSION`. This can be a glob expression.",
+            example = "jackson-module*")
+    String artifactId;
+
+
+
+    @Override
+    public String getDisplayName() {
+        return "Remove a Gradle dependency";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Removes a single dependency from the dependencies section of the `build.gradle`.";
+    }
+
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getApplicableTest() {
```

Comment:
getSingleSourceApplicableTest is more appropriate for this one, as it will cause the recipe to run only on those source files that are gradle scripts.

Reply:
Done :)